src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1280,6 +1281,16 @@ namespace winrt::TerminalApp::implementation
             auto guidWString = Utils::GuidToString(profile.Guid());
 
             StringMap envMap{};
+            if (_settings.GlobalSettings().ReloadEnvironmentVariables())
+            {
+                til::env environment;
+                environment.regenerate();
+
+                for (const auto& [key, value] : environment.as_map())
+                {
+                    envMap.Insert(key, value);
+                }
+            }
             envMap.Insert(L"WT_PROFILE_ID", guidWString);
```

Comment:
I love the simplicity of this, but I am a little bit concerned about what happens after we hand this environment block off.

We produce a new env block here, painstakingly inserting it into envMap... and then ConptyConnection uses `EnvironmentVariableMapW` to _generate a new map based on the current process environment_ and inserts the contents of `envMap` from here into it.

In brief... what we will get is a hybrid of Terminal's spawning environment and the current environment. As well, we will probably end up parsing the entire env twice.

Should we move this down to ConptyConnection (worm the boolean down in through the valueset for all I mind, ha) and make sure we do not call `UpdateEnvironmentMapW`?